pkg/engine/virtualmachines.go

```diff
@@ -405,6 +422,21 @@ func createAgentAvailabilitySetVM(cs *api.ContainerService, profile *api.AgentPo
 		}
 	}
 
+	for _, addonProfile := range addonProfiles {
+		if addonProfile.Enabled && addonProfile.Identity != nil && addonProfile.Identity.ResourceID != "" {
+			if virtualMachine.Identity == nil {
+				virtualMachine.Identity = &compute.VirtualMachineIdentity{}
+				virtualMachine.Identity.Type = compute.ResourceIdentityTypeUserAssigned
+				virtualMachine.Identity.UserAssignedIdentities = make(map[string]*compute.VirtualMachineIdentityUserAssignedIdentitiesValue)
+			} else if virtualMachine.Identity.Type == compute.ResourceIdentityTypeSystemAssigned {
```

Comment:
Aren't there other possibilities such as:
1. Identity.Type is None ?
2. Identity.Type is already ResourceIdentityTypeUserAssigned? 

Can we please handle/add comments on how those scenarios works ?

Reply:
Thanks for reminding, will add comment.